src/p4est_communication.h

```diff
@@ -575,6 +575,141 @@ void                p4est_transfer_items_end (p4est_transfer_context_t * tc);
  */
 void                p4est_transfer_end (p4est_transfer_context_t * tc);
 
+/** Callback function for \ref p4est_transfer_search.
+ * 
+ * \param[in] which_tree Tree containing quadrant
+ * \param[in] quadrant The quadrant
+ * \param[in] point The point
+ * \param[in] user Optional user defined context
+ * \return True if \a point intersects \a quadrant.
+ */
+typedef int         (*p4est_intersect_t) (p4est_topidx_t which_tree,
+                                          p4est_quadrant_t *quadrant,
+                                          void *point, void *user);
+
+/** The points being exchanged in \ref p4est_transfer_search. */
+typedef struct p4est_transfer_search {
+  /* The points to exchange */
+  sc_array_t *points;
+
+  /* Each process is responsible for propagating the first num_resp points that
+   * it knows. The remaining points are additionally known to other processes
+   * which are responsible for their propagation. */
+  p4est_locidx_t num_resp;
+
+} p4est_transfer_search_t;
+
+/** Destroy a \ref p4est_transfer_search_t structure
+ * 
+ * \param [in,out] c the structure to destroy
+ */
+void p4est_transfer_search_destroy (p4est_transfer_search_t *c);
```

Comment:
We may remove this function and instead provide a full documentation of the transfer_search_t on input and output of the transfer_search functions.